mm2src/lp_swap/recreate_swap_data.rs

```diff
@@ -157,6 +160,8 @@ fn recreate_maker_swap(ctx: MmArc, taker_swap: TakerSavedSwap) -> RecreateSwapRe
         taker_pubkey: started_event.my_persistent_pub,
         maker_coin_swap_contract_addr: negotiated_event.maker_coin_swap_contract_addr,
         taker_coin_swap_contract_addr: negotiated_event.taker_coin_swap_contract_addr,
+        maker_coin_htlc_pubkey: None,
+        taker_coin_htlc_pubkey: None,
```

Comment:
Should these be `None`? doesn't the `MakerNegotiationData` have this info?

Reply:
`MakerNegotiationData` has the pubkeys of the maker. The taker has different keys, which should be derived from his `maker_coin_htlc_privkey` and `taker_coin_htlc_privkey`. But these fields are reset on swap status broadcast, I can add `maker_coin_htlc_pubkey` and `taker_coin_htlc_pubkey` to `TakerSwapData` and `MakerSwapData` - we will be able to recreate more of essential swap data using them.